internal/backend/runtime/omni/controllers/omni/cluster_machine_config.go

```diff
@@ -468,13 +468,13 @@ func buildInstallImage(imageFactoryHost string, resID resource.ID, installImage
 
 	schematicID := installImage.SchematicId
 
-	secureBootStatus := installImage.SecureBootStatus
-	if secureBootStatus == nil { // should never happen - must have been handled before entering this function
+	bootType := installImage.BootType
+	if bootType == nil { // should never happen - must have been handled before entering this function
 		return "", fmt.Errorf("machine %q has no secure boot status set", resID)
 	}
 
 	installerName := "installer"
-	if secureBootStatus.Enabled {
+	if bootType.IsSecureBoot {
 		installerName = "installer-secureboot"
```

Comment:
@smira should we get back into having a unified installer from factory, so we could simplify this?

Reply:
yes, we should probably at least create an issue for it